From b2f247e1bdc85cd06c19eccd580eae9501ee0235 Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Fri, 5 Aug 2022 09:47:52 +0200
Subject: [PATCH] Explicitly disable zstd support

Make sure we don't pick up a libzstd.so dependency if it happens
to be installed on the system.
---
 src/bootstrap/native.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/bootstrap/native.rs b/src/bootstrap/native.rs
index 4d548dbb63875..8b50d5dc52bc1 100644
--- a/src/bootstrap/native.rs
+++ b/src/bootstrap/native.rs
@@ -325,6 +325,9 @@ impl Step for Llvm {
             cfg.define("LLVM_PROFDATA_FILE", &path);
         }
 
+        // Disable zstd to avoid a dependency on libzstd.so.
+        cfg.define("LLVM_ENABLE_ZSTD", "OFF");
+
         if target != "aarch64-apple-darwin" && !target.contains("windows") {
             cfg.define("LLVM_ENABLE_ZLIB", "ON");
         } else {
